### PR TITLE
Add lineage_job_namespace and lineage_job_name OpenLineage macros

### PR DESCRIPTION
**Pull request by dolfinus**
Add new `lineage_job_namespace()` and `lineage_job_name(task_instance)` macros to OpenLineage provider. Same as https://github.com/OpenLineage/OpenLineage/pull/2582.

Also updated macros documentation a bit.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/openlineage/plugins/macros.py [hunk 1]
```diff
@@ -26,22 +26,44 @@
...
-    This can be used to forward the run id from a task to a child run so the job hierarchy is preserved.
+    This can be used to forward the runId from a task to a child run so the job hierarchy is preserved.
 
     .. seealso::
-        For more information on how to use this operator, take a look at the guide:
+        For more information take a look at the guide:
         :ref:`howto/macros:openlineage`
     """
     if TYPE_CHECKING:
         assert task_instance.task
```
Comment (kacpermuda) [inline airflow/providers/openlineage/plugins/macros.py]: I think we don't need this anymore, since we stopped using `task_instance.task`

**Review by kacpermuda — APPROVED** [review]
LGTM

File: airflow/providers/openlineage/plugins/macros.py (hunk 1 above)
Comment (dolfinus) [reply airflow/providers/openlineage/plugins/macros.py]: Fixed

**Review by mobuchowski — APPROVED** [review]
Thanks @dolfinus for the contribution.

**Pull request merged**